langchain/vectorstores/matching_engine.py

```diff
@@ -0,0 +1,433 @@
+
+        Args:
+            data: The data that will be stored.
+            gcs_location: The location where the data will be stored.
+        """
+        bucket = self.gcs_client.get_bucket(self.gcs_bucket_name)
+        blob = bucket.blob(gcs_location)
+        blob.upload_from_string(data)
+
+    def similarity_search(
+        self, query: str, k: int = 4, **kwargs: Any
+    ) -> List[Document]:
+        """Return docs most similar to query.
+
+        Args:
+            query: The string that will be used to search for similar documents.
+            k: The amount of neighbors that will be retrieved.
+
+        Returns:
+            A list of k matching documents.
+        """
+
+        logger.debug(f"Embedding query {query}.")
+        embedding_query = self.embedding.embed_documents([query])
+
+        response = self.endpoint.match(
+            deployed_index_id=self._get_index_id(),
+            queries=embedding_query,
+            num_neighbors=k,
+        )
+
+        if len(response) == 0:
+            return []
+
+        logger.debug(f"Found {len(response)} matches for the query {query}.")
+
+        results = []
+
+        # I'm only getting the first one because queries receives an array
+        # and the similarity_search method only recevies one query. This
+        # means that the match method will always return an array with only
+        # one element.
+        for doc in response[0]:
+            page_content = self._download_from_gcs(f"documents/{doc.id}")
+            results.append(Document(page_content=page_content))
+
+        logger.debug("Downloaded documents for query.")
+
+        return results
+
+    def _get_index_id(self) -> str:
+        """Gets the correct index id for the endpoint.
+
+        Returns:
+            The index id if found (which should be found) or throws
+            ValueError otherwise.
+        """
+        for index in self.endpoint.deployed_indexes:
+            if index.index == self.index.resource_name:
+                return index.id
+
+        raise ValueError(
+            f"No index with id {self.index.resource_name} "
+            f"deployed on enpoint "
+            f"{self.endpoint.display_name}."
+        )
+
+    def _download_from_gcs(self, gcs_location: str) -> str:
+        """Downloads from GCS in text format.
+
+        Args:
+            gcs_location: The location where the file is located.
+
+        Returns:
+            The string contents of the file.
+        """
+        bucket = self.gcs_client.get_bucket(self.gcs_bucket_name)
+        blob = bucket.blob(gcs_location)
+        return blob.download_as_string()
+
+    @classmethod
+    def from_texts(
+        cls: Type["MatchingEngine"],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> "MatchingEngine":
+        """Use from components instead."""
+        raise NotImplementedError(
```

Comment:
@hwchase17 why was this removed? Previously, it would have worked regardless if the developer called `from_texts` or `from_components`

Reply:
think the from_texts interface doesn't fit super well, since it only allows texts and embedding as args, whereas the init and from_components both expect a number of args. we could technically pipe all kwargs for from_text into one of those two but it's pretty hacky. also there's no real value to using from_texts for this class, since it'd just be wrapping instantiation and calling add_texts